implement property-qos propagate getter and setters

trittsv opened the pull request:
This PR implements getter and setter for property propagate on discovery.

Previously it was not possible to propagate the Property-Qos on the network for the c++ and python wrappers.

@eboasson could you have a look?

eboasson commented on src/core/ddsc/include/dds/ddsc/dds_public_qos.h:
```diff
@@ -482,6 +482,23 @@ dds_qset_prop (
+ * @returns - false if any of the arguments is invalid, the qos is not present in the qos object or there was no property found with the provided name
+ */
+DDS_EXPORT bool
+dds_qset_prop_propagate (
```
I would instead add a function `dds_qset_prop_propagate(dds_qos_t *qos, const char *name, const char *value, bool propagate)` and define `dds_qset_prop` using that function: `dds_qset_prop(qos, name, value) { dds_qset_prop_propagate(qos, name, value, false); }`. I think that's a lot easier than having to set the properties by name and then set the propagation.

eboasson commented on src/core/ddsc/include/dds/ddsc/dds_public_qos.h:
```diff
@@ -1010,6 +1027,23 @@ dds_qget_prop (
   const char * name,
   char ** value);
 
+/**
```
Here I would follow the same pattern as what I suggested for the setter: a `dds_qget_prop_propagate` that *also* returns the setting of the "propagate" flag (if a non-null pointer is passed in), then implement `dds_qget_prop` as:
```
bool dds_qget_prop (const dds_qos_t *qos, const char *name, char **value)
{
  return dds_qget_prop_propagate (qos, name, value, NULL);
}
```

(And again the same for the binary property one.)

trittsv commented:
@eboasson thanks a lot for the review, i updated the functions as you noted and added also the binary properties propagate functions. Furthermore it includes your patch so that the default participant qos are always set.

May you can have a look again? :)
